newrelic-alerts-configurator/src/main/java/com/ocadotechnology/newrelic/alertsconfigurator/NrqlConditionConfigurator.java

```diff
@@ -35,17 +34,24 @@
 
     @Override
     protected AlertsNrqlCondition convertFromConfigItem(NrqlCondition condition) {
-        return AlertsNrqlCondition.builder()
-                .name(condition.getConditionName())
-                .enabled(condition.isEnabled())
-                .runbookUrl(condition.getRunBookUrl())
-                .terms(TermsUtils.createNrqlTerms(condition.getTerms()))
-                .valueFunction(condition.getValueFunction().getValueString())
-                .nrql(Nrql.builder()
-                        .sinceValue(String.valueOf(condition.getSinceValue().getSince()))
-                        .query(condition.getQuery())
-                        .build())
-                .build();
+        AlertsNrqlCondition.AlertsNrqlConditionBuilder nrqlConditionBuilder = AlertsNrqlCondition.builder()
+            .name(condition.getConditionName())
+            .enabled(condition.isEnabled())
+            .runbookUrl(condition.getRunBookUrl())
+            .terms(TermsUtils.createNrqlTerms(condition.getTerms()))
+            .valueFunction(condition.getValueFunction().getValueString())
+            .nrql(Nrql.builder()
+                .sinceValue(String.valueOf(condition.getSinceValue().getSince()))
+                .query(condition.getQuery())
+                .build());
+
+        condition.getSignal()
+            .ifPresent(nrqlSignalConfiguration -> nrqlConditionBuilder
+                .signal(SignalUtils.createSignal(nrqlSignalConfiguration))
+                .expiration(ExpirationUtils.createExpiration(nrqlSignalConfiguration.getSignalLostConfiguration()))
```

Comment:
I'm wondering why New Relic decided to design API in the way that `signal` and `expiration` are on the same level, next to each other.
Your approach with `SignalLostConfiguration` being part of `NrqlSignalConfiguration` seems more intuitive.

Maybe `expiration` can be configured without `signal` (or with some other option)? Have you checked it?

On the other hand, I saw a lot of really stupid decisions in the New Relic API, so it maybe just the thing they did without any good reason

Reply:
Yes, `expiration` and `signal` could be configured separately (default values will be applied), but in fact, both properties describe behaviour of signal – that's why I decided it is better to "merge" them in one configuration for user of library api.